spec/02-integration/03-admin_api/06-cluster_routes_spec.lua

```diff
@@ -88,17 +88,21 @@ describe("Admin API", function()
         })
         assert.res_status(200, res) -- why not 204??
 
-        ngx.sleep(3)
+        local json
+
+        helpers.wait_until(function()
+          res = assert(client:send {
+            method = "GET",
+            path = "/cluster"
+          })
+          local body = assert.res_status(200, res)
+          json = cjson.decode(body)
+          assert.equal(2, #json.data)
+          assert.equal(2, json.total)
+          assert.equal("alive", json.data[1].status)
+          return json.data[2].status == "leaving"
+        end, 10)
```

Comment:
to improve error/testfailure reporting we could update the `wait_until` function to receive from the callback `success, err`, in case of a timeout, the last reported `err` will be appended. 

eg. Instead of error `timeout`, it would give `timeout: row not found error` or whatever.

If then wrapping the callback in a `pcall` assertions can be used. Such that it could return;
`timeout: assertion failed expected 2, got xyz`.